localstack-core/localstack/services/events/provider.py

```diff
@@ -339,33 +338,112 @@ def _create_connection_arn(
         self, context: RequestContext, name: str, connection_uuid: str
     ) -> str:
         """Create a standardized connection ARN."""
-        return f"arn:aws:events:{context.region}:{context.account_id}:connection/{name}/{connection_uuid}"
+        return f"arn:{get_partition(context.region)}:events:{context.region}:{context.account_id}:connection/{name}/{connection_uuid}"
+
+    def _get_secret_value(
+        self,
+        authorization_type: ConnectionAuthorizationType,
+        auth_parameters: CreateConnectionAuthRequestParameters,
+    ) -> str:
+        result = {}
+        match authorization_type:
+            case ConnectionAuthorizationType.BASIC:
+                params = auth_parameters.get("BasicAuthParameters", {})
+                result = {"username": params.get("Username"), "password": params.get("Password")}
+            case ConnectionAuthorizationType.API_KEY:
+                params = auth_parameters.get("ApiKeyAuthParameters", {})
+                result = {
+                    "api_key_name": params.get("ApiKeyName"),
+                    "api_key_value": params.get("ApiKeyValue"),
+                }
+            case ConnectionAuthorizationType.OAUTH_CLIENT_CREDENTIALS:
+                params = auth_parameters.get("OAuthParameters", {})
+                client_params = params.get("ClientParameters", {})
+                result = {
+                    "client_id": client_params.get("ClientID"),
+                    "client_secret": client_params.get("ClientSecret"),
+                    "authorization_endpoint": params.get("AuthorizationEndpoint"),
+                    "http_method": params.get("HttpMethod"),
+                }
 
-    def _create_secret_arn(self, context: RequestContext, name: str, connection_uuid: str) -> str:
+        if "InvocationHttpParameters" in auth_parameters:
+            result["invocation_http_parameters"] = auth_parameters["InvocationHttpParameters"]
+
+        return json.dumps(result)
+
+    def _create_connection_secret(
+        self,
+        context: RequestContext,
+        name: str,
+        authorization_type: ConnectionAuthorizationType,
+        auth_parameters: CreateConnectionAuthRequestParameters,
+    ) -> str:
         """Create a standardized secret ARN."""
-        return f"arn:aws:secretsmanager:{context.region}:{context.account_id}:secret:events!connection/{name}/{connection_uuid}"
+        # TODO use service role as described here: https://docs.aws.amazon.com/eventbridge/latest/userguide/using-service-linked-roles-service-action-1.html
+        # not too important as it is created automatically on AWS anyway, with the right permissions
```

Comment:
even when using the  API or only via  the AWS Console?

Reply:
Should also be the case with the api.